addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
I didn't say scroll was sync, I said setting scrollTop was sync, which generates an almost immediate scroll event. The following pattern should always work:

```
function updateScrollAndWaitForEvent(element, value) {
   return new Promise(resolve => {
    element.scrollTop = value;
    setTimeout(resolve, 1);
  });
}
```